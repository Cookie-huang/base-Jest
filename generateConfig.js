export const generateConfig = () => {
  return {
    server: "http://localhost",
    port: "8080",
    version: 1,
    time: new Date()
  };
};

export const generateAnotherConfig = () => {
  return {
    server: "http://localhost",
    port: "8080",
    version: 2
  };
};

/* w -> u    快照全部更新 */
/* w -> i -> u   快照交互式更新 */
