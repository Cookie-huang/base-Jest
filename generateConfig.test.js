import { generateConfig, generateAnotherConfig } from "./generateConfig";

test.only("快照1", () => {
  expect(generateConfig()).toMatchInlineSnapshot(
    {
      time: expect.any(Date),
    },
    `
    Object {
      "port": "8080",
      "server": "http://localhost",
      "time": Any<Date>,
      "version": 1,
    }
  `
  );
});

test.only("快照2", () => {
  expect(generateAnotherConfig()).toMatchSnapshot({
    // time: jest.Any(Date)
  });
});
