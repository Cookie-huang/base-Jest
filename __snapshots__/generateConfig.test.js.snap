// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`快照2 1`] = `
Object {
  "port": "8080",
  "server": "http://localhost",
  "version": 2,
}
`;
